rest/route/build_task.go

```diff
@@ -105,17 +106,20 @@ func (tbh *tasksByBuildHandler) Run(ctx context.Context) gimlet.Responder {
 
 	tasks = tasks[:lastIndex]
 	for i := range tasks {
+		taskCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
 		taskModel := &model.APITask{}
 
-		if err = taskModel.BuildFromService(ctx, &tasks[i], &model.APITaskArgs{
+		if err = taskModel.BuildFromService(taskCtx, &tasks[i], &model.APITaskArgs{
 			IncludeAMI:               true,
 			IncludeArtifacts:         true,
 			IncludeProjectIdentifier: true,
 			LogURL:                   tbh.url,
 			ParsleyLogURL:            tbh.parsleyURL,
 		}); err != nil {
+			cancel()
```

Comment:
While this is correct code, it's generally a sign of a code smell if you need to cancel manually (except in a few specific cases) instead of defer cancel. It's preferred to instead put the loop body in a helper function and `defer cancel()` in that function to make sure the context cancels at the end of the one iteration. This is slightly for organizational purposes, but more so to avoid a potential bug where you forget to cancel in one of the early exit cases. Something like:
```go
for i := range tasks {
    taskModel, err := getTheTaskData(...)
    resp.AddData(taskModel)
}

...

func getTheTaskData(...) {
    ctx, cancel := context.WithTimeout(...)
    defer cancel()

    // Actually get task data.
}
```